server/exporter.js

```diff
@@ -15,45 +19,188 @@ const queueOptions = {
   },
 };
 
-async function exportAndEmailCollections(teamId: string, email: string) {
+async function fileOperationsUpdate(teamId, userId, exportData) {
+  await Event.add({
+    name: "fileOperations.update",
+    teamId: teamId,
+    actorId: userId,
+    data: {
+      type: exportData.type,
+      id: exportData.id,
+      state: exportData.state,
+      size: exportData.size,
+      collectionId: exportData.collectionId,
+      createdAt: exportData.createdAt,
+    },
+  });
+}
+
+type exportAndEmailCollectionsType = {|
+  teamId: string,
+  userId: string,
+  email: string,
+  collectionId?: string,
+  fileOperationId?: string,
+|};
+
+async function exportAndEmailCollections({
+  teamId,
+  userId,
+  email,
+  collectionId,
+  fileOperationId,
+}: exportAndEmailCollectionsType) {
   log("Archiving team", teamId);
   const { archiveCollections } = require("./utils/zip");
   const team = await Team.findByPk(teamId);
-  const collections = await Collection.findAll({
-    where: { teamId },
-    order: [["name", "ASC"]],
-  });
+  const user = await User.findByPk(userId);
+
+  let collections;
+  if (!collectionId) {
+    collections = await Collection.scope({
```

Comment:
There is a convenience method for this, `await user.collectionIds();`, that calculates all of the permissions. So could grab the id's and then query collections with id in that list.